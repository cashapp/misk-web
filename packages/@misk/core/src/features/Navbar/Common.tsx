import { css } from "@emotion/core"
import { defaultTheme, Environment, ITheme } from "src/utilities"

export const cssNavbarHeading = (theme: ITheme) => css`
  font-size: 24px !important;
  text-decoration: none;
  text-transform: uppercase;
  letter-spacing: 0px;
  margin-right: 30px;
  color: ${theme.navbarText};
  min-width: fit-content;

  a {
    color: ${theme.navbarText} !important;
    letter-spacing: 1px;
    text-decoration: none;
    &:hover {
      color: ${theme.navbarLinkHover} !important;
      text-decoration: none;
    }
  }
`

export const cssNavbarHeadingEnvironment = (
  environment: Environment,
  theme: ITheme = defaultTheme
) =>
  css(
    cssNavbarHeading(theme),
    `
color: ${theme.environmentToColor(environment)} !important;
min-width: 0;
`
  )

export const cssMiskLink = (theme: ITheme) => css`
  color: ${theme.navbarText};
  text-decoration: none;
  &:hover {
    color: ${theme.navbarLinkHover};
    text-decoration: none;
  }
`
