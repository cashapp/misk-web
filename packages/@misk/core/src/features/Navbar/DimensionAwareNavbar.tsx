/** @jsx jsx */
import { Alignment, Navbar, NavbarGroup } from "@blueprintjs/core"
import { css, jsx } from "@emotion/core"
import * as React from "react"
import { Link } from "react-router-dom"
import { ResponsiveContainer } from "../../cssContainers"
import {
  Banner,
  HomeLink,
  IBannerExternalProps,
  IDimensionAwareProps,
  IMenuExternalProps,
  Menu,
  processNavbarItems,
  calculateNavbarItemsByScreenWidth
} from "../Navbar"
import { Environment } from "../../utilities"
import { ITheme, defaultTheme } from "src/utilities/theme"

/**
 * <DimensionAwareComponent
 *    height={this.state.height}
 *    width={this.state.width}
 *    environment={this.props.environment}
 *    environmentBannerVisible={this.props.environmentBannerVisible}
 *    environmentNavbarVisible={this.props.environmentBannerVisible}
 *    error={this.props.error}
 *    homeName={this.props.homeName}
 *    homeUrl={this.props.homeUrl}
 *    links={this.props.links}
 *    navbarItems={this.props.navbarItems}
 *    status={this.props.status}
 *    theme={this.props.theme}
 *  />
 */

export interface INavbarProps
  extends IBannerExternalProps,
    IMenuExternalProps,
    IThemeProps {
  environmentNavbarVisible?: Environment[]
  homeName?: string | Element | JSX.Element
  homeUrl?: string
  navbar_items?: Array<string | Element | JSX.Element>
  navbarItemsToDisplay?: number
}

export interface IThemeProps {
  theme?: ITheme
}

const cssNavbar = (theme: ITheme) => css`
  background-color: ${theme.navbarBackground} !important;
  padding-top: 10px !important;
  padding-bottom: 60px !important;
  position: fixed !important;
`

const cssNavbarGroup = (theme: ITheme) => css`
  font-size: 13px !important;
  font-weight: 600 !important;
  position: relative;
  padding-top: 25px;
  padding-bottom: 27px;
  @media (max-width: 870px) {
    padding-left: 60px;
  }
  @media (min-width: 992px) and (max-width: 1085px) {
    padding-left: 60px;
  }
  @media (min-width: 1200px) and (max-width: 1285px) {
    padding-left: 60px;
  }
`

export class DimensionAwareNavbar extends React.Component<
  IDimensionAwareProps & INavbarProps,
  {}
> {
  // TODO see if this can be removed
  public state = {
    isOpen: false
  }

  public render() {
    const {
      environment,
      environmentBannerVisible,
      environmentNavbarVisible,
      error,
      homeName,
      homeUrl,
      linkComponent = Link,
      links,
      menuIcon,
      menuOpenIcon,
      menuButtonAsLink,
      menuShowButton,
      navbar_items,
      navbarItemsToDisplay = null,
      status,
      theme = defaultTheme,
      width
    } = this.props
    const processedNavbarItems = processNavbarItems(
      theme,
      environment,
      environmentNavbarVisible,
      navbar_items
    )
    const numberOfNavbarItemsToDisplay =
      navbarItemsToDisplay == null
        ? calculateNavbarItemsByScreenWidth(width)
        : navbarItemsToDisplay
    return (
      <Navbar css={cssNavbar(theme)}>
        <ResponsiveContainer>
          <NavbarGroup
            align={Alignment.LEFT}
            className="bp3-dark"
            css={cssNavbarGroup(theme)}
          >
            <HomeLink
              linkComponent={linkComponent}
              homeName={homeName}
              homeUrl={homeUrl}
              theme={theme}
            />
            {processedNavbarItems.slice(0, numberOfNavbarItemsToDisplay)}
          </NavbarGroup>
        </ResponsiveContainer>
        <Menu
          processedNavbarItems={processedNavbarItems}
          error={error}
          links={links}
          linkComponent={linkComponent}
          menuIcon={menuIcon}
          menuOpenIcon={menuOpenIcon}
          menuButtonAsLink={menuButtonAsLink}
          menuShowButton={menuShowButton}
          theme={theme}
        />
        <Banner
          environment={environment}
          environmentBannerVisible={environmentBannerVisible}
          status={status}
          theme={theme}
        />
      </Navbar>
    )
  }
}
