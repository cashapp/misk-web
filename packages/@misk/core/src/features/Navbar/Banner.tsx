/** @jsx jsx */
import { css, jsx } from "@emotion/core"
import * as React from "react"
import { TextHTMLOrElementComponent } from "../../components"
import { FlexContainer, ResponsiveContainer } from "../../cssContainers"
import { Environment, ITheme, defaultTheme } from "../../utilities"
import { IThemeProps } from "./DimensionAwareNavbar"

/**
 * <Banner
 *    environment={this.props.environment}
 *    environmentBannerVisible={this.props.environmentBannerVisible}
 *    status={this.props.status}
 *  />
 */

export interface IBannerExternalProps extends IThemeProps {
  environment?: Environment
  environmentBannerVisible?: Environment[]
  status?: string | Element | JSX.Element
}

const cssNavbarBanner = (environment: Environment, theme: ITheme) => css`
  background-color: ${theme.environmentToColor(environment)} !important;
  color: ${theme.bannerText} !important;
  text-align: center;
  font-weight: 600;
  padding: 5px 10px;
  position: fixed !important;
  width: 100%;
  top: 70px;
  left: 0px;
  z-index: 1010 !important;

  a {
    font-weight: 300;
    color: ${theme.bannerText};
    text-decoration: underline;
    letter-spacing: 1px;
    &:hover {
      color: ${theme.bannerLinkHover};
      text-decoration: underline;
    }
  }
`

export class Banner extends React.Component<IBannerExternalProps, {}> {
  public render() {
    const {
      environment,
      environmentBannerVisible,
      status,
      theme = defaultTheme
    } = this.props
    if (
      environmentBannerVisible &&
      environmentBannerVisible.includes(environment)
    ) {
      return (
        <span css={cssNavbarBanner(environment, theme)}>
          <ResponsiveContainer>
            <FlexContainer>
              <TextHTMLOrElementComponent>{status}</TextHTMLOrElementComponent>
            </FlexContainer>
          </ResponsiveContainer>
        </span>
      )
    } else {
      return <div />
    }
  }
}
