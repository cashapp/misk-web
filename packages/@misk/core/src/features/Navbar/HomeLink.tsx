/** @jsx jsx */
import { css, jsx } from "@emotion/core"
import { Link } from "react-router-dom"
import { cssMiskLink, cssNavbarHeading } from "./Common"
import { IThemeProps } from "./DimensionAwareNavbar"
import { defaultTheme } from "src/utilities"

/**
 * <HomeLink
 *    homeName={this.props.homeName}
 *    homeUrl={this.props.homeUrl}
 *  />
 */

export interface IHomeLinkProps extends IThemeProps {
  homeName?: string | Element | JSX.Element
  homeUrl?: string
  linkComponent?: any
}

export const HomeLink = (props: IHomeLinkProps) => {
  const { homeName, homeUrl, theme = defaultTheme } = props
  const LinkComponent = props.linkComponent || Link
  if (homeName && homeUrl) {
    return (
      <LinkComponent
        css={css(cssMiskLink(theme), { minWidth: "fit-content" })}
        to={homeUrl}
      >
        <span css={cssNavbarHeading(theme)}>{homeName}</span>
      </LinkComponent>
    )
  } else if (homeName) {
    return <span css={cssNavbarHeading(theme)}>{homeName}</span>
  } else {
    return <span css={cssNavbarHeading(theme)}>Misk</span>
  }
}
