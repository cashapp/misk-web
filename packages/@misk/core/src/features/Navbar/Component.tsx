import { IResizeEntry, ResizeSensor } from "@blueprintjs/core"
import reduce from "lodash/reduce"
import * as React from "react"
import {
  defaultEnvironment,
  defaultEnvironmentIndicatorsVisible
} from "../../utilities"
import { DimensionAwareNavbar, INavbarProps } from "../Navbar"
import { defaultTheme } from "src/utilities/theme"

/**
 * <Component
 *    environment={environment}
 *    environmentBannerVisible={[Environment.DEVELOPMENT]}
 *    environmentNavbarVisible={[Environment.DEVELOPMENT]}
 *    error={error}
 *    homeName={"Misk Home"}
 *    homeUrl={"/"}
 *    links={links}
 *    menuIcon={IconNames.FOLDER_CLOSE}
 *    menuOpenIcon={IconNames.FOLDER_OPEN}
 *    menuButtonAsLink={true}
 *    menuShowButton={true}
 *    navbarItems={[ "Test1", '<a href="#">Test2</>', <span key={2}>Test3</span> ]}
 *    status={"News Item"}
 *  />
 */

export interface IDimensionAwareProps {
  height: number
  width: number
}

export class Navbar extends React.Component<INavbarProps, {}> {
  public state = {
    height: 0,
    width: 0
  }

  handleResize = (entries: IResizeEntry[]) => {
    this.setState(
      reduce(
        entries,
        (dimension, e) => ({
          height: Math.max(dimension.height, e.contentRect.height),
          width: Math.max(dimension.width, e.contentRect.width)
        }),
        { height: 0, width: 0 }
      )
    )
  }

  public render() {
    const {
      environment = defaultEnvironment,
      environmentBannerVisible = defaultEnvironmentIndicatorsVisible,
      environmentNavbarVisible = defaultEnvironmentIndicatorsVisible,
      error,
      homeName,
      homeUrl,
      linkComponent,
      links,
      menuIcon,
      menuOpenIcon,
      menuButtonAsLink,
      menuShowButton,
      navbar_items,
      navbarItemsToDisplay = null,
      status,
      theme = defaultTheme
    } = this.props
    const { height, width } = this.state
    return (
      <ResizeSensor onResize={this.handleResize}>
        <DimensionAwareNavbar
          height={height}
          width={width}
          environment={environment}
          environmentBannerVisible={environmentBannerVisible}
          environmentNavbarVisible={environmentNavbarVisible}
          error={error}
          homeName={homeName}
          homeUrl={homeUrl}
          linkComponent={linkComponent}
          links={links}
          menuIcon={menuIcon}
          menuOpenIcon={menuOpenIcon}
          menuButtonAsLink={menuButtonAsLink}
          menuShowButton={menuShowButton}
          navbar_items={navbar_items}
          navbarItemsToDisplay={navbarItemsToDisplay}
          status={status}
          theme={theme}
        />
      </ResizeSensor>
    )
  }
}
