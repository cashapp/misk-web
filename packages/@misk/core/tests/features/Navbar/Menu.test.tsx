import React from "react"
import { BrowserRouter } from "react-router-dom"
import { cleanup, render } from "@testing-library/react"
import { Menu } from "src/features/Navbar"
import { adminDashboardTabs, axiosError } from "../../testUtilities"

describe("Navbar Menu", () => {
  afterEach(cleanup)
  it("Menu can render", () => {
    const { asFragment } = render(<Menu />)
    expect(asFragment()).toMatchSnapshot()
  })
  it("Menu can render with error", () => {
    const { asFragment } = render(<Menu error={axiosError} />)
    expect(asFragment()).toMatchSnapshot()
  })
  it("Menu can render with links", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Menu links={adminDashboardTabs} />
      </BrowserRouter>
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("Menu can render with rendered navbar items", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Menu
          processedNavbarItems={[
            <span>{"Navbar Item Alpha"}</span>,
            <span>{"Navbar Item Bravo"}</span>
          ]}
        />
      </BrowserRouter>
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("Menu can render with links and rendered navbar items", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Menu
          links={adminDashboardTabs}
          processedNavbarItems={[
            <span>{"Navbar Item Alpha"}</span>,
            <span>{"Navbar Item Bravo"}</span>
          ]}
        />
      </BrowserRouter>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
