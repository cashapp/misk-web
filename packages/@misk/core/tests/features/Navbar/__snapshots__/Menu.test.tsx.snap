// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Menu Menu can render 1`] = `
<DocumentFragment>
  .emotion-1 {
  background-color: #29333a !important;
  box-shadow: none !important;
  background-image: none !important;
  top: 15px;
  left: 15px;
  position: absolute;
  z-index: 1020;
}

.emotion-0 {
  color: #7a7e82 !important;
}

.emotion-0:hover {
  color: white;
}

.emotion-6 {
  background-color: #29333a;
  display: block;
  margin: 60px -20px 0 -20px;
}

.emotion-5 {
  min-height: 250px;
  padding: 50px 0px;
  overflow-y: scroll;
  max-height: 100vh;
}

@media (max-width:768px) {
  .emotion-5 {
    padding: 50px 20px;
  }
}

.emotion-4 {
  margin: 0 auto;
}

@media (min-width:768px) {
  .emotion-4 {
    width: 750px;
  }
}

@media (min-width:992px) {
  .emotion-4 {
    width: 970px;
  }
}

@media (min-width:1200px) {
  .emotion-4 {
    width: 1170px;
  }
}

.emotion-3 {
  display: inline-block;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-2 * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<div>
    <button
      class="bp3-button emotion-1"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        <span
          class="bp3-icon bp3-icon-menu emotion-0"
          icon="menu"
        >
          <svg
            data-icon="menu"
            height="32"
            viewBox="0 0 20 20"
            width="32"
          >
            <desc>
              menu
            </desc>
            <path
              d="M1 6h18c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm18 3H1c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1zm0 5H1c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </span>
    </button>
    <div
      class="emotion-6"
    >
      <div
        class="bp3-collapse"
      >
        <div
          aria-hidden="true"
          class="bp3-collapse-body"
          style="transform: translateY(-0px);"
        >
          <div
            class="emotion-5"
          >
            <div
              class="emotion-4"
            >
              <div
                class="emotion-3"
              >
                <div
                  class="emotion-2"
                />
              </div>
              <span />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Navbar Menu Menu can render with error 1`] = `
<DocumentFragment>
  .emotion-1 {
  background-color: #29333a !important;
  box-shadow: none !important;
  background-image: none !important;
  top: 15px;
  left: 15px;
  position: absolute;
  z-index: 1020;
}

.emotion-0 {
  color: #7a7e82 !important;
}

.emotion-0:hover {
  color: white;
}

.emotion-8 {
  background-color: #29333a;
  display: block;
  margin: 60px -20px 0 -20px;
}

.emotion-7 {
  min-height: 250px;
  padding: 50px 0px;
  overflow-y: scroll;
  max-height: 100vh;
}

@media (max-width:768px) {
  .emotion-7 {
    padding: 50px 20px;
  }
}

.emotion-6 {
  margin: 0 auto;
}

@media (min-width:768px) {
  .emotion-6 {
    width: 750px;
  }
}

@media (min-width:992px) {
  .emotion-6 {
    width: 970px;
  }
}

@media (min-width:1200px) {
  .emotion-6 {
    width: 1170px;
  }
}

.emotion-3 {
  display: inline-block;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-2 * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.emotion-5 {
  margin: 20px 0;
}

.emotion-4 {
  text-align: left;
}

<div>
    <button
      class="bp3-button emotion-1"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        <span
          class="bp3-icon bp3-icon-menu emotion-0"
          icon="menu"
        >
          <svg
            data-icon="menu"
            height="32"
            viewBox="0 0 20 20"
            width="32"
          >
            <desc>
              menu
            </desc>
            <path
              d="M1 6h18c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm18 3H1c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1zm0 5H1c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </span>
    </button>
    <div
      class="emotion-8"
    >
      <div
        class="bp3-collapse"
      >
        <div
          aria-hidden="true"
          class="bp3-collapse-body"
          style="transform: translateY(-0px);"
        >
          <div
            class="emotion-7"
          >
            <div
              class="emotion-6"
            >
              <div
                class="emotion-3"
              >
                <div
                  class="emotion-2"
                />
              </div>
              <div
                class="bp3-callout bp3-intent-danger bp3-callout-icon emotion-5"
              >
                <span
                  class="bp3-icon bp3-icon-error"
                  icon="error"
                >
                  <svg
                    data-icon="error"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <desc>
                      error
                    </desc>
                    <path
                      d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm1 16H9v-2h2v2zm0-3H9V4h2v9z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                <h4
                  class="bp3-heading"
                >
                  [401]  Not authorized. 
                </h4>
                <pre
                  class="emotion-4"
                >
                  {
  "response": {
    "status": 401,
    "statusText": "Not authorized",
    "data": ""
  }
}
                </pre>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Navbar Menu Menu can render with links 1`] = `
<DocumentFragment>
  .emotion-1 {
  background-color: #29333a !important;
  box-shadow: none !important;
  background-image: none !important;
  top: 15px;
  left: 15px;
  position: absolute;
  z-index: 1020;
}

.emotion-0 {
  color: #7a7e82 !important;
}

.emotion-0:hover {
  color: white;
}

.emotion-11 {
  background-color: #29333a;
  display: block;
  margin: 60px -20px 0 -20px;
}

.emotion-10 {
  min-height: 250px;
  padding: 50px 0px;
  overflow-y: scroll;
  max-height: 100vh;
}

@media (max-width:768px) {
  .emotion-10 {
    padding: 50px 20px;
  }
}

.emotion-9 {
  margin: 0 auto;
}

@media (min-width:768px) {
  .emotion-9 {
    width: 750px;
  }
}

@media (min-width:992px) {
  .emotion-9 {
    width: 970px;
  }
}

@media (min-width:1200px) {
  .emotion-9 {
    width: 1170px;
  }
}

.emotion-3 {
  display: inline-block;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-2 * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-bottom: 35px;
}

.emotion-8 * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.emotion-4 {
  color: #cecece;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 16px;
  -webkit-flex-basis: 300px;
  -ms-flex-preferred-size: 300px;
  flex-basis: 300px;
  padding: 5px 0;
  color: #cecece;
}

.emotion-4:hover {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
    <button
      class="bp3-button emotion-1"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        <span
          class="bp3-icon bp3-icon-menu emotion-0"
          icon="menu"
        >
          <svg
            data-icon="menu"
            height="32"
            viewBox="0 0 20 20"
            width="32"
          >
            <desc>
              menu
            </desc>
            <path
              d="M1 6h18c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm18 3H1c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1zm0 5H1c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </span>
    </button>
    <div
      class="emotion-11"
    >
      <div
        class="bp3-collapse"
      >
        <div
          aria-hidden="true"
          class="bp3-collapse-body"
          style="transform: translateY(-0px);"
        >
          <div
            class="emotion-10"
          >
            <div
              class="emotion-9"
            >
              <div
                class="emotion-3"
              >
                <div
                  class="emotion-2"
                />
              </div>
              <div>
                <span />
                <div
                  class="emotion-8"
                >
                  <a
                    class="emotion-4"
                    href="/first-tab/"
                  >
                    First Tab
                  </a>
                  <a
                    class="emotion-4"
                    href="https://github.com/cashapp/misk-web"
                  >
                    Misk Web Repo
                  </a>
                  <a
                    class="emotion-4"
                    href="/palette/"
                  >
                    Palette
                  </a>
                  <a
                    class="emotion-4"
                    href="/second-tab/"
                  >
                    Second Tab
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Navbar Menu Menu can render with links and rendered navbar items 1`] = `
<DocumentFragment>
  .emotion-1 {
  background-color: #29333a !important;
  box-shadow: none !important;
  background-image: none !important;
  top: 15px;
  left: 15px;
  position: absolute;
  z-index: 1020;
}

.emotion-0 {
  color: #7a7e82 !important;
}

.emotion-0:hover {
  color: white;
}

.emotion-11 {
  background-color: #29333a;
  display: block;
  margin: 60px -20px 0 -20px;
}

.emotion-10 {
  min-height: 250px;
  padding: 50px 0px;
  overflow-y: scroll;
  max-height: 100vh;
}

@media (max-width:768px) {
  .emotion-10 {
    padding: 50px 20px;
  }
}

.emotion-9 {
  margin: 0 auto;
}

@media (min-width:768px) {
  .emotion-9 {
    width: 750px;
  }
}

@media (min-width:992px) {
  .emotion-9 {
    width: 970px;
  }
}

@media (min-width:1200px) {
  .emotion-9 {
    width: 1170px;
  }
}

.emotion-3 {
  display: inline-block;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-2 * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-bottom: 35px;
}

.emotion-8 * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.emotion-4 {
  color: #cecece;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 16px;
  -webkit-flex-basis: 300px;
  -ms-flex-preferred-size: 300px;
  flex-basis: 300px;
  padding: 5px 0;
  color: #cecece;
}

.emotion-4:hover {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
    <button
      class="bp3-button emotion-1"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        <span
          class="bp3-icon bp3-icon-menu emotion-0"
          icon="menu"
        >
          <svg
            data-icon="menu"
            height="32"
            viewBox="0 0 20 20"
            width="32"
          >
            <desc>
              menu
            </desc>
            <path
              d="M1 6h18c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm18 3H1c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1zm0 5H1c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </span>
    </button>
    <div
      class="emotion-11"
    >
      <div
        class="bp3-collapse"
      >
        <div
          aria-hidden="true"
          class="bp3-collapse-body"
          style="transform: translateY(-0px);"
        >
          <div
            class="emotion-10"
          >
            <div
              class="emotion-9"
            >
              <div
                class="emotion-3"
              >
                <div
                  class="emotion-2"
                >
                  <span>
                    <span>
                      Navbar Item Alpha
                    </span>
                  </span>
                  <span>
                    <span>
                      Navbar Item Bravo
                    </span>
                  </span>
                </div>
              </div>
              <div>
                <span />
                <div
                  class="emotion-8"
                >
                  <a
                    class="emotion-4"
                    href="/first-tab/"
                  >
                    First Tab
                  </a>
                  <a
                    class="emotion-4"
                    href="https://github.com/cashapp/misk-web"
                  >
                    Misk Web Repo
                  </a>
                  <a
                    class="emotion-4"
                    href="/palette/"
                  >
                    Palette
                  </a>
                  <a
                    class="emotion-4"
                    href="/second-tab/"
                  >
                    Second Tab
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Navbar Menu Menu can render with rendered navbar items 1`] = `
<DocumentFragment>
  .emotion-1 {
  background-color: #29333a !important;
  box-shadow: none !important;
  background-image: none !important;
  top: 15px;
  left: 15px;
  position: absolute;
  z-index: 1020;
}

.emotion-0 {
  color: #7a7e82 !important;
}

.emotion-0:hover {
  color: white;
}

.emotion-6 {
  background-color: #29333a;
  display: block;
  margin: 60px -20px 0 -20px;
}

.emotion-5 {
  min-height: 250px;
  padding: 50px 0px;
  overflow-y: scroll;
  max-height: 100vh;
}

@media (max-width:768px) {
  .emotion-5 {
    padding: 50px 20px;
  }
}

.emotion-4 {
  margin: 0 auto;
}

@media (min-width:768px) {
  .emotion-4 {
    width: 750px;
  }
}

@media (min-width:992px) {
  .emotion-4 {
    width: 970px;
  }
}

@media (min-width:1200px) {
  .emotion-4 {
    width: 1170px;
  }
}

.emotion-3 {
  display: inline-block;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-2 * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<div>
    <button
      class="bp3-button emotion-1"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        <span
          class="bp3-icon bp3-icon-menu emotion-0"
          icon="menu"
        >
          <svg
            data-icon="menu"
            height="32"
            viewBox="0 0 20 20"
            width="32"
          >
            <desc>
              menu
            </desc>
            <path
              d="M1 6h18c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm18 3H1c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1zm0 5H1c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </span>
    </button>
    <div
      class="emotion-6"
    >
      <div
        class="bp3-collapse"
      >
        <div
          aria-hidden="true"
          class="bp3-collapse-body"
          style="transform: translateY(-0px);"
        >
          <div
            class="emotion-5"
          >
            <div
              class="emotion-4"
            >
              <div
                class="emotion-3"
              >
                <div
                  class="emotion-2"
                >
                  <span>
                    <span>
                      Navbar Item Alpha
                    </span>
                  </span>
                  <span>
                    <span>
                      Navbar Item Bravo
                    </span>
                  </span>
                </div>
              </div>
              <span />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
